core/scheduler/service/executor_input_compiler.go

```diff
@@ -62,6 +63,33 @@ type InputCompiler struct {
 	logger log.Logger
 }
 
+func sanitiseLabel(key string) string {
+	// implements validation from https://cloud.google.com/bigquery/docs/labels-intro#requirements
+
+	// max length 63 chars
+	// if length greater than 63 then truncate to 61 and add __ at end to denote truncation
+	if maxLabelLength := 63; len(key) > maxLabelLength {
+		key = key[len(key)-61:] + "__"
+	}
+	// Keys can contain only lowercase letters, numeric characters, underscores, and dashes.
+	key = strings.ToLower(key)
+	re := regexp.MustCompile(`[^\w-]`)
+	key = re.ReplaceAllString(key, "-")
+	return key
+}
+
+func getJobLabelsString(labels map[string]string) string {
```

Comment:
could we reuse this function https://github.com/goto/optimus/blob/a1a0aa1093325a8a6710a2a82cdd3b3d733042e2/core/scheduler/job.go#L133 ?

Reply:
this will introduce these labels in a many other places, increasing regression 

although we can choose the sanitise the existing labels too to avoid them causing failures in BQ